lib/pure/asyncmacro.nim

```diff
@@ -257,7 +178,7 @@ proc asyncSingleProc(prc: NimNode): NimNode {.compileTime.} =
     outerProcBody.add(prc.body[0])
 
   # -> var retFuture = newFuture[T]()
-  var retFutureSym = genSym(nskVar, "retFuture")
+  var retFutureSym = ident"retFuture"
```

Comment:
Why was this changed? Without `genSym` here if someone declares a `retFuture` of their own then there will be a clash...

Reply:
in order for await to be able to check for this symbol in when IIRC

Reply:
Ahh. Yeah, I'm not the biggest fan of this. If someone accidentally defines a `retFuture` in their proc then it'll be considered async (that behaviour will be very confusing).

I wonder if we can come up with a better way to detect async procs, can you check if `compiles(yield f)` works? :)

Reply:
interesting idea, but this would work in another iterator as well !  which we still don't want. 

maybe i can generate a symbol which can't be written by users similar to `:env`

Reply:
> but this would work in another iterator as well ! which we still don't want.

IMO it's not the worst thing if it does work, it'll only work if that iterator's return type is a future and if a future is awaited, so likelihood is that it will make sense anyway.

Reply:
it is far fetched, but it's possible to write a sync iterator returning futures: it makes sense in some cases

Reply:
also, i am a bit scared of the `compiles(yield f)` performance thing: for every await, wouldn't that slow down things a bit

Reply:
we can do something else : we can just leave out `template await` and and generate just one in of `async`: Future[T] working and one in multisync sync: which works for every type: all other cases just fail like normal overloads!

Reply:
this way we can really just use the default nim mechanisms for most of the errors as well (await just isn't defined somewhere, or it just doesnt accepts int somewhere)

Reply:
and if we really want to we can still leave a global overload which errors with a more detailed message as a fallback

Reply:
this works!

Reply:
i'll push a new commit when i test it a bit